tensorboard/components/vz_example_viewer/vz-example-viewer.ts

```diff
@@ -276,8 +275,20 @@ Polymer({
   },
 
   getFilteredFeaturesList: function(featureList: NameAndFeature[],
-      searchValue: string) {
+      searchValue: string, saliency: SaliencyMap) {
     let filtered = featureList;
+    const checkSal = saliency && Object.keys(saliency).length > 0;
+    // Create a dict of feature names to the total absolute saliency of all
+    // its feature values, to sort features with the most salienct features at
+    // the top.
+    const saliencyTotals = checkSal ?
+         Object.assign({}, ...Object.keys(saliency).map(
```

Comment:
This is doing `Object.assign` on objects that you map out of keys of `saliency`. I think you can do this either by cloning saliency object and transforming its values or 
```
Object.entries(saliency)
	.map(([key, value]) => [key, typeof value == 'number' ? [value] : value])
	.map(([key, value]) => [key, value.map(Math.abs).reduce((total, cur) => total + cur, 0)])
        .reduce((obj, [key, value]) => {
           obj[key] = value;
           return obj;
        }, {})
```
though I am not sure if this has any better readability.